frontend/apps/filemanager/filemanager.lua

```diff
@@ -596,134 +611,255 @@ function FileManager:onSwipeFM(ges)
+                    callback = function()
+                        self:goHome()
                         UIManager:close(self.file_dialog)
-                        if ok then
-                            self.file_chooser:changeToPath(sd_path)
-                        end
-                    else
+                    end
+                }
+            },
+            {
+                {
+                    text = _("Open random document"),
+                    callback = function()
+                        self:openRandomFile(self.file_chooser.path)
                         UIManager:close(self.file_dialog)
-                        self.file_chooser:changeToPath(Device.home_dir)
                     end
-                end,
+                }
             },
-        })
+            {
+                {
+                    text = _("Folder shortcuts"),
+                    callback = function()
+                        self:handleEvent(Event:new("ShowFolderShortcutsDialog"))
+                        UIManager:close(self.file_dialog)
+                    end
+                }
+            }
+        }
+
+        if Device:canImportFiles() then
+            table.insert(buttons, 4, {
+                {
+                    text = _("Import files here"),
+                    enabled = Device:isValidPath(self.file_chooser.path),
+                    callback = function()
+                        local current_dir = self.file_chooser.path
+                        UIManager:close(self.file_dialog)
+                        Device.importFile(current_dir)
+                    end,
+                },
+            })
+        end
+
+        if Device:hasExternalSD() then
+            table.insert(buttons, 5, {
+                {
+                    text_func = function()
+                        if Device:isValidPath(self.file_chooser.path) then
+                            return _("Switch to SDCard")
+                        else
+                            return _("Switch to internal storage")
+                        end
+                    end,
+                    callback = function()
+                        if Device:isValidPath(self.file_chooser.path) then
+                            local ok, sd_path = Device:hasExternalSD()
+                            UIManager:close(self.file_dialog)
+                            if ok then
+                                self.file_chooser:changeToPath(sd_path)
+                            end
+                        else
+                            UIManager:close(self.file_dialog)
+                            self.file_chooser:changeToPath(Device.home_dir)
+                        end
+                    end,
+                },
+            })
+        end
     end
 
     self.file_dialog = ButtonDialogTitle:new{
-        title = BD.dirpath(filemanagerutil.abbreviate(self.file_chooser.path)),
+        title = title,
         title_align = "center",
+        use_info_style = true,
```

Comment:
Why change this ? for the "2 selected files" display that should have less importance ?
For the regular display, I prefer seeing the filename in bold than not.
Before:
![image](https://user-images.githubusercontent.com/24273478/145446114-502771d4-70a6-457a-8448-11c182c55ab3.png)

After (I find the contrast between non-bold title and bold buttons a bit odd):
![image](https://user-images.githubusercontent.com/24273478/145446170-f87908bd-3fe8-4d90-9293-10c1d77c184b.png)

Reply:
Yes, actually a matter of taste.
I like your second screenshot emphasizing the difference between the header and the buttons.
Also https://github.com/koreader/koreader/pull/7767#issuecomment-850353248. Can we have more opinions?

Reply:
I agree with @hius07, personally I feel we overuse the bold text

Reply:
> I prefer seeing the filename in bold than not

Single file operations menu is unchanged. Regular font is used for the Plus menu only.

![14](https://user-images.githubusercontent.com/62179190/145455407-f666ffbf-b63a-4c98-be2c-9858abb35ab4.png)

Reply:
In the OPDS dialog, it is plain info text being no longer bold (that I may not mind).
In here, it's a title, _centered_, making it more like a heading / dialog title (and there, the missing bold feel more odd to me that in the OPDS dialog).

Reply:
I've tested it on my Kobo - and it doesn't look that bad :) It gets a noticably larger font size (more noticably larger that what I saw on the emulator), which makes it quite alright, different, and readable. (Like on your screenshot just above.)
It could give somehow a new distinguishable identity to these dialogs where the title is a filename or folder.
The question is whether this new identitiy is not too excentric :) and if it's alright to have _some_ titles not in bold where everything else is in bold.

(If you go on with this, in the CoverBrowser hacks, pick it from the thing we hack like we do with `local orig_title_align = self.file_dialog.title_align` and `title_align = orig_title_align`.)

Reply:
> Single file operations menu is unchanged. Regular font is used for the Plus menu only.

That might be ok for some incremental change to get @Frenzie to tell us what he thinks :)
But having the tap menu with a dirname big and non bold, and single file long-press dialog with a filename smaller and bold, sounds like in..con...sis....ten...cy.

Reply:
I'd be inclined to make the header bold but I'm sure it's fine either way.